dmoj/graders/interactive.py

```diff
@@ -0,0 +1,77 @@
+from dmoj.graders.standard import StandardGrader
+
+class WrongAnswer(BaseException):
+    pass
+
+class Interactor(object):
+    def __init__(self, process):
+        self.process = process
+        self._tokens = None
+
+    def read(self):
+        return self.process.stdout.read()
+
+    def readln(self, strip_newline=True):
+        if strip_newline:
```

Comment:
This code duplication is awful, please store the `readline` into a variable and then optionally `rstrip` on it.